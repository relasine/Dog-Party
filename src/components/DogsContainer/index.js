import React from 'react';
import PropTypes from 'prop-types';
import './DogsContainer.css';
import DogCard from '../DogCard';

import doggo from '../../assets/dog2-sq.jpg';
import anotherDoggo from '../../assets/dog3-sq.jpg';
import dogTheThird from '../../assets/dog4-sq.jpg';

const DogsContainer = ({ category }) => {
  const allDogs = [
  {title: 'How To Dog', image: doggo, text: 'Dat tungg tho pupperino length boy. borkdrive aqua doggo. Maximum borkdrive long doggo extremely cuuuuuute you are doing me the shock borkf, heckin good boys doggo doge. The neighborhood pupper borkf smol corgo long water shoob stop it fren, boofers pupper heck ruff.'},
  {title: 'What Is Dogs', image: anotherDoggo, text: 'Long doggo smol borking doggo with a long snoot for pats pupperino h*ck heckin angery woofer shoober aqua doggo heck, doggo aqua doggo most angery pupper I have ever seen heckin angery woofer pupper. ruff adorable doggo very hand that feed shibe. Extremely cuuuuuute corgo waggy wags stop it fren.'},
  {title: 'Facts From Dog', image: dogTheThird, text: 'Most angery pupper I have ever seen such treat heck, dat tungg tho. You are doin me a concern shibe such treat ruff heck, stop it fren ur givin me a spook. Puggorino long water shoob dat tungg tho very hand that feed shibe long bois he made many woofs.'}
  ];

  let dogs;

  if (category === 'All Dogs') {
    dogs = allDogs.map(dog => {
      return (
        <DogCard dog={dog} key={dog.title}/>
      )
    });
  } else {
    const theDog = allDogs.find(dog => {
      return category === dog.title;
    });

    dogs = <DogCard dog={theDog} key={theDog.title}/>;
  }

  return (
    <section>
      {dogs}
    </section>
  )
}

DogsContainer.propTypes = {
  category: PropTypes.string.isRequired
};

export default DogsContainer;