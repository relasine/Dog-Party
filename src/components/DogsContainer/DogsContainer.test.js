import React from 'react';
import ReactDOM from 'react-dom';
import DogsContainer from './index.js';
import { shallow } from 'enzyme';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<DogsContainer dogs={[]}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('should match the snapshot', () => {
  const wrapper = shallow(<DogsContainer dogs={[]}/>);

  expect(wrapper).toMatchSnapshot();
});