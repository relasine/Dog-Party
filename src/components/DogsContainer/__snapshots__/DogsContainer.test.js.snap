// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<section>
  <DogCard
    dog={
      Object {
        "image": "dog2-sq.jpg",
        "text": "Dat tungg tho pupperino length boy. borkdrive aqua doggo. Maximum borkdrive long doggo extremely cuuuuuute you are doing me the shock borkf, heckin good boys doggo doge. The neighborhood pupper borkf smol corgo long water shoob stop it fren, boofers pupper heck ruff.",
        "title": "How To Dog",
      }
    }
    key="How To Dog"
  />
  <DogCard
    dog={
      Object {
        "image": "dog3-sq.jpg",
        "text": "Long doggo smol borking doggo with a long snoot for pats pupperino h*ck heckin angery woofer shoober aqua doggo heck, doggo aqua doggo most angery pupper I have ever seen heckin angery woofer pupper. ruff adorable doggo very hand that feed shibe. Extremely cuuuuuute corgo waggy wags stop it fren.",
        "title": "What Is Dogs",
      }
    }
    key="What Is Dogs"
  />
  <DogCard
    dog={
      Object {
        "image": "dog4-sq.jpg",
        "text": "Most angery pupper I have ever seen such treat heck, dat tungg tho. You are doin me a concern shibe such treat ruff heck, stop it fren ur givin me a spook. Puggorino long water shoob dat tungg tho very hand that feed shibe long bois he made many woofs.",
        "title": "Facts From Dog",
      }
    }
    key="Facts From Dog"
  />
</section>
`;
